DataConnectors/WorkplaceFacebook/WorkplaceWebhooksTrigger/__init__.py

```diff
@@ -0,0 +1,105 @@
+import logging
+import hashlib
+import hmac
+import os
+import azure.functions as func
+import json
+import re
+import base64
+import requests
+import datetime
+
+
+AppSecret = os.environ['WorkplaceAppSecret']
+VerifyToken = os.environ['WorkplaceVerifyToken']
+customer_id = os.environ['WorkspaceID']
+shared_key = os.environ['WorkspaceKey']
+logAnalyticsUri = os.environ.get('logAnalyticsUri')
+log_type = 'Workplace_Facebook'
+
+if ((logAnalyticsUri in (None, '') or str(logAnalyticsUri).isspace())):
+    logAnalyticsUri = 'https://' + customer_id + '.ods.opinsights.azure.com'
+pattern = r'https:\/\/([\w\-]+)\.ods\.opinsights\.azure.([a-zA-Z\.]+)$'
+match = re.match(pattern,str(logAnalyticsUri))
+if(not match):
+    raise Exception("Workplace_Facebook: Invalid Log Analytics Uri.")
+
+
+def hmac_sha1(message, secret):
+    message = bytes(message, 'utf-8')
+    secret = bytes(secret, 'utf-8')
+    hash = hmac.new(secret, message, hashlib.sha1)
+    return hash.hexdigest()
+
+
+def parse_signature(value):
+    parts = value.split('&')
+    ret = {}
+    for kv in parts:
+        (k, v) = kv.split('=')
+        ret[k] = v
+    return ret
+
+
+def build_signature(customer_id, shared_key, date, content_length, method, content_type, resource):
+    x_headers = 'x-ms-date:' + date
+    string_to_hash = method + "\n" + str(content_length) + "\n" + content_type + "\n" + x_headers + "\n" + resource
+    bytes_to_hash = bytes(string_to_hash, encoding="utf-8")
+    decoded_key = base64.b64decode(shared_key)
+    encoded_hash = base64.b64encode(hmac.new(decoded_key, bytes_to_hash, digestmod=hashlib.sha256).digest()).decode()
+    authorization = "SharedKey {}:{}".format(customer_id,encoded_hash)
+    return authorization
+
+
+def post_data_to_sentinel(body):
+    method = 'POST'
+    content_type = 'application/json'
+    resource = '/api/logs'
+    rfc1123date = datetime.datetime.utcnow().strftime('%a, %d %b %Y %H:%M:%S GMT')
+    content_length = len(body)
+    signature = build_signature(customer_id, shared_key, rfc1123date, content_length, method, content_type, resource)
+    uri = 'https://' + customer_id + '.ods.opinsights.azure.com' + resource + '?api-version=2016-04-01'

```

Comment:
Please use **logAnalyticsUri** instead of  'https://' + customer_id + '.ods.opinsights.azure.com'.

Reply:
@averbn I have pushed this change. Please check. Thanks!